docs/api/modulus.models.rst

```diff
@@ -2,6 +2,403 @@
+
+        def forward(self, x):
+            x1 = self.enc1(x)
+            x2 = self.enc2(x1)
+            x = self.dec1(x2)
+            return self.final(x)
+
+Now we show this model rewritten in Modulus. First, let's subclass the model from 
+``modulus.models.module.Module`` instead of ``torch.nn.Module``. The
+``modulus.models.module.Module`` class acts like a direct replacement for the
+``torch.nn.Module`` and provides additional functionality for saving and loading
+checkpoints, etc. Refer to the API docs of ``modulus.models.module.Module`` for further
+details. Additionally we will add metadata to the model to capture the optimizations
+enabled. In this case we will enable CUDA Graphs and Automatic Mixed-Precision.
+
+.. code:: python
+
+    from dataclasses import dataclass
+    from modulus.models.meta import ModelMetaData
+    from modulus.models.module import Module 
+    import torch.nn as nn
+
+    @dataclass
+    class MetaData(ModelMetaData):
+        name: str = "UNet"
+        # Optimization
+        jit: bool = False
+        cuda_graphs: bool = True
+        amp_cpu: bool = True
+        amp_gpu: bool = True
+    
+    class UNet(Module):
+        def __init__(self, in_channels=1, out_channels=1):
+            super(UNet, self).__init__(meta=MetaData())
+
+            self.enc1 = self.conv_block(in_channels, 64)
+            self.enc2 = self.conv_block(64, 128)
+
+            self.dec1 = self.upconv_block(128, 64)
+            self.final = nn.Conv2d(64, out_channels, kernel_size=1)
+
+        def conv_block(self, in_channels, out_channels):
+            return nn.Sequential(
+                nn.Conv2d(in_channels, out_channels, 3, padding=1),
+                nn.ReLU(inplace=True),
+                nn.MaxPool2d(2)
+            )
+
+        def upconv_block(self, in_channels, out_channels):
+            return nn.Sequential(
+                nn.ConvTranspose2d(in_channels, out_channels, 2, stride=2),
+                nn.Conv2d(out_channels, out_channels, 3, padding=1),
+                nn.ReLU(inplace=True)
+            )
+
+        def forward(self, x):
+            x1 = self.enc1(x)
+            x2 = self.enc2(x1)
+            x = self.dec1(x2)
+            return self.final(x)
+
+Now that we have our Modulus model, we can make use of these optimizations using the
+``modulus.utils.StaticCaptureTraining`` decorator. This decorator will capture the
+training step function and optimize it for the specified optimizations.
+
+.. code:: python
+
+    import torch
+    from modulus.utils import StaticCaptureTraining
+
+    model = UNet().to("cuda")
+    input = torch.randn(8, 1, 128, 128).to("cuda")
+    output = torch.zeros(8, 1, 64, 64).to("cuda")
+
+    optim = torch.optim.Adam(model.parameters(), lr=0.001)
+
+    # Create training step function with optimization wrapper
+    @StaticCaptureTraining(
+        model=model,
+        optim=optim,
+        cuda_graph_warmup=11,
+    )
+    def training_step(invar, outvar):
```

Comment:
Added some comments here on this if you think its good enough. Would be really nice to have this be a much longer section but given that it might change I think this is fine.